src/reconcile/rec_write.c

```diff
@@ -221,13 +221,16 @@ __reconcile(WT_SESSION_IMPL *session, WT_REF *ref, WT_SALVAGE_COOKIE *salvage, u
 #ifdef HAVE_DIAGNOSTIC
     addr = ref->addr;
 #endif
-    /* Wrap up the page reconciliation. */
-    if (ret == 0 && (ret = __rec_write_wrapup(session, r, page)) == 0)
-        __rec_write_page_status(session, r);
-    else {
+    if (ret != 0) {
         /* Make sure that reconciliation doesn't free the page that has been written to disk. */
         WT_ASSERT(session, addr == NULL || ref->addr != NULL);
         WT_TRET(__rec_write_wrapup_err(session, r, page));
+    } else {
+        /* Wrap up the page reconciliation. */
+        ret = __rec_write_wrapup(session, r, page);
+        if (ret != 0)
+            WT_RET_PANIC(session, ret, "failed to wrap up reconciliation");
+        __rec_write_page_status(session, r);
```

Comment:
I'm not panic on `__rec_root_write`. This function recursively calls into `__wt_reconcile` so it will panic if it hits the same issue. However, it can throw errors in other cases as well and I'm unsure if I should panic or not in those cases.

The final call to `__wt_page_parent_modify_set` may fail on malloc but also I'm not sure we should panic on that or not.

@luke-pearson @bvpvamsikrishna

Reply:
In my opinion we should panic on any failure after writing the page image. We may want to rename `__rec_write_wrapup_err` as its naming implies it should be called after `__rec_write_wrapup`. But for now I think its okay until we figure out the details of this change.

Reply:
Before this change, it may called after `__rec_write_wrapup`. With this change, I agree it should be renamed.